### PR TITLE
feat: show user message on retired email search

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-2521](https://openedx.atlassian.net/browse/PROD-2521)
Screenshot 2022-03-10 at 2 18 37 AM

For the retired email, we weren't able to get anything from Support Tools hence we changed the message here so that it explains why the account info associated to the email cannot be received.

### CODE REVIEW

File: src/users/data/api.js [hunk 1]
```diff
@@ -108,7 +108,16 @@ export async function getUser(userIdentifier) {
...
+    if (JSON.parse(error.customAttributes.httpErrorResponseData)) {
+      error.userError = {
+        code: null,
+        dismissible: true,
+        text: JSON.parse(error.customAttributes.httpErrorResponseData).error_msg,
+        type: 'error',
+        topic: 'general',
+      };
+      throw error;
+    } else if (isEmail(userIdentifier)) {
```
Comment (azanbinzahid) [inline src/users/data/api.js]: I think we should just set `notFoundErrorText` here and follow the existing flow. Error is thrown after `if/else` block.

**Review by azanbinzahid — CHANGES_REQUESTED** [review]

File: src/users/data/api.js [hunk 2]
```diff
@@ -108,7 +108,9 @@ export async function getUser(userIdentifier) {
     console.log(JSON.parse(error.customAttributes.httpErrorResponseData));
 
     let notFoundErrorText = null;
-    if (isEmail(userIdentifier)) {
+    if (JSON.parse(error.customAttributes.httpErrorResponseData)) {
+      notFoundErrorText = JSON.parse(error.customAttributes.httpErrorResponseData).error_msg;
```
Comment (azanbinzahid) [inline src/users/data/api.js]: One more thing, you may need to check if these attrib exist or not, just to prevent any unexpected error.

**Review by azanbinzahid — APPROVED** [review]

File: src/users/data/api.js [hunk 3]
```diff
@@ -108,7 +108,9 @@ export async function getUser(userIdentifier) {
     console.log(JSON.parse(error.customAttributes.httpErrorResponseData));
 
     let notFoundErrorText = null;
-    if (isEmail(userIdentifier)) {
+    if (JSON.parse(error.customAttributes?.httpErrorResponseData)) {
```
Comment (DawoudSheraz) [inline src/users/data/api.js:111]: ?. what is this syntax?

File: src/users/data/api.test.js [hunk 4]
```diff
@@ -451,6 +452,22 @@ describe('API', () => {
...
+        topic: 'general',
+      };
+      mockAdapter.onGet(`${userAccountApiBaseUrl}?email=${encodeURIComponent(testEmail)}`).reply(() => throwError(404, { error_msg: messages.USER_RETIRED_EMAIL_IDENTIFIER_ERROR }));
+      try {
+        await api.getUser(testEmail);
+      } catch (error) {
+        expect(error.userError).toEqual(expectedUserError);
+      }
+    });
+
```
Comment (DawoudSheraz) [inline src/users/data/api.test.js]: A bit difficult to read here. Add the comment on why the error handling is needed here.

File: src/users/data/api.js (hunk 3 above)
Comment (Ali-D-Akbar) [reply src/users/data/api.js:111]: if `customAttributes` exists (isn't false), get `httpErrorResponseData`.

Comment (DawoudSheraz) [reply src/users/data/api.js:111]: Is this Es6 syntax? Seeing this for the first time.

**Review by DawoudSheraz — APPROVED** [review]
one small nit comment, otherwise good to go.

File: src/users/data/api.js (hunk 3 above)
Comment (Ali-D-Akbar) [reply src/users/data/api.js:111]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

File: src/users/data/api.test.js (hunk 4 above)
Comment (Ali-D-Akbar) [reply src/users/data/api.test.js]: improved readability and added a comment too.

**Pull request merged**